OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs

```diff
@@ -39,7 +40,7 @@ public override IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInit
 					var haveNeighbour = false;
 					foreach (var n in kv.Value)
 					{
-						var rb = init.World.Map.Rules.Actors[n].TraitInfoOrDefault<WithWallSpriteBodyInfo>();
+						var rb = init.World.Map.Rules.Actors[n].TraitInfos<WithWallSpriteBodyInfo>().FirstOrDefault(Exts.IsTraitEnabled);
```

Comment:
Apologies for throwing up more roadblocks for getting this merged, but we're going to need an `IWallConnectorInfo` interface here to make sure that walls connect to gates in the map editor.

I'd also be happy to defer this to a followup PR.